app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -401,6 +392,10 @@ export default class TopicController extends Controller {
       return;
     }
     event?.preventDefault();
+    const selection = window.getSelection();
```

Comment:
This is a weird thing for us to be doing, so I think it would be good to add a comment

```suggestion
    // If the user is selecting the title, don't go into edit mode
    const selection = window.getSelection();
```